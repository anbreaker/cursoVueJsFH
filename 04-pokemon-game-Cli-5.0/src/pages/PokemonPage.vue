<template>
  <div>
    <div v-if="!pokemon">
      <h1>Loading...</h1>
    </div>
    <div v-else>
      <h1>Who is this Pokemon?</h1>

      <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
      <PokemonOptions :pokemons="pokemonArr" @selectionPokemon="checkAnswer" />

      <div class="center" v-if="showAnswer">
        <h2>{{ message }}</h2>
        <button @click="newGame">New Game</button>
      </div>
    </div>
  </div>
</template>

<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'

import getPokemonOptions from '@/helpers/getPokemonOptions'

export default {
  name: 'PokemonPage',
  components: {
    PokemonPicture,
    PokemonOptions,
  },

  data() {
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false,
      showAnswer: false,
      message: '',
    }
  },

  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions()

      const rndInt = Math.floor(Math.random() * 4)

      this.pokemon = this.pokemonArr[rndInt]
    },

    checkAnswer(selectedId) {
      this.showPokemon = true
      this.showAnswer = true

      if (selectedId === this.pokemon.id)
        this.message = `Correct!, is ${this.pokemon.name}`
      else
        this.message = `Opps!, the correct pokemon is ${this.pokemon.name.replace(
          /^\w/,
          (c) => c.toUpperCase()
        )}`
    },

    newGame() {
      this.pokemonArr = []
      this.showPokemon = false
      this.showAnswer = false
      this.pokemon = null
      this.mixPokemonArray()
    },
  },

  mounted() {
    this.mixPokemonArray()
  },
  beforeUnmount() {
    this.newGame()
  },
}
</script>

<style scoped>
h1,
h2 {
  display: flex;
  justify-content: center;
  color: aliceblue;
}
button {
  display: block;
  margin: 0 auto;
}
</style>
