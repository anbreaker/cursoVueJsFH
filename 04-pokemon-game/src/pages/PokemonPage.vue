<template>
  <div>
    <h1>Who is this Pokemon?</h1>

    <PokemonPicture :pokemonId="1" :showPokemon="true" />
    <PokemonOptions />
  </div>
</template>

<script>
import PokemonOptions from "@/components/PokemonOptions";
import PokemonPicture from "@/components/PokemonPicture";

export default {
  name: "PokemonPage",
  components: {
    PokemonPicture,
    PokemonOptions
  }
};
</script>

<style lang="scss" scoped>
</style>