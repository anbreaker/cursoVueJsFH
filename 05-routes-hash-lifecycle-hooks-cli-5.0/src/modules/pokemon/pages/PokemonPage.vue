<template>
  <div>
    <h1>
      Pokemon Page
      <span>#{{ id }}</span>
    </h1>
    <div v-if="pokemon">
      <img :src="pokemon.sprites.front_default" :alt="pokemon.name" />
    </div>
  </div>
</template>

<script>
  export default {
    props: {
      id: {
        type: Number,
        required: true,
      },
    },

    data() {
      return {
        pokemon: null,
      };
    },

    created() {
      this.getPokemon();
    },

    methods: {
      async getPokemon() {
        try {
          const pokemon = await fetch(`https://pokeapi.co/api/v2/pokemon/${this.id}`);

          this.pokemon = await pokemon.json();
        } catch (error) {
          this.$router.push('/');
          console.log('Error: ', error, '<------');
        }
      },
    },
    watch: {
      id() {
        console.log(this.id);
        this.getPokemon();
      },
    },
  };
</script>
