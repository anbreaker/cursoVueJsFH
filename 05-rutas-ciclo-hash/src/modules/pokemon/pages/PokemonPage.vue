<template>
  <div>
    <h1>
      Pokemon Page with id <span>#{{ id }}</span>
    </h1>

    <div v-if="pokemon">
      <img :src="pokemon.sprites.front_default" alt="pokemon.name" />
    </div>
  </div>
</template>

<script>
export default {
  props: {
    id: {
      type: Number,
      required: true
    }
  },
  data() {
    return {
      // id: this.$route.params.id,
      pokemon: null
    };
  },

  created() {
    // const { id } = this.$route.params;
    // this.id = id;
    // console.log(this.id);
    // console.log(this.$attrs);
    this.getPokemon();
  },

  methods: {
    async getPokemon() {
      try {
        this.pokemon = await fetch(
          `https://pokeapi.co/api/v2/pokemon/${this.id}`
        ).then(response => response.json());

        // console.log(this.pokemon);
      } catch (error) {
        console.log(error);
        this.$router.push("/");
        console.log("Id not correct!");
      }
    }
  },
  watch: {
    id() {
      console.log("entra el wtch");
      console.log(this.id);
      this.getPokemon();
    }
  }
};
</script>