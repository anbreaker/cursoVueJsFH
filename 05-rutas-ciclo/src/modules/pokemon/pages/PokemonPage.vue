<template>
  <h1>Pokemon Page with id #{{ id }}</h1>
</template>

<script>
export default {
  data() {
    return {
      id: null
    };
  },
  created() {
    const { id } = this.$route.params;
    this.id = id;

    console.log(this.id);
  }
};
</script>