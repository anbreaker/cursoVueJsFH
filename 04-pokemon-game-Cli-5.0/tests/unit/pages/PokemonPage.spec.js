import { shallowMount, mount } from '@vue/test-utils'

import PokemonPage from '@/pages/PokemonPage'
import { pokemonArr } from '../mocks/pokemons.mock'

describe('PokemonPage Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage)
  })
  test('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('must call mixPokemonArray on Mount', () => {
    const mixPokemonArraySpy = jest.spyOn(
      PokemonPage.methods,
      'mixPokemonArray'
    )

    // to mount component after call methods
    shallowMount(PokemonPage)

    expect(mixPokemonArraySpy).toHaveBeenCalled()
  })

  test('Should match the snapshot, when pokemons is loaded', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArr,
          pokemon: pokemonArr[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
        }
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Must show the components PokemonPicture and PokemonOptions', () => {
    const wrapper = shallowMount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemonArr,
          pokemon: pokemonArr[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
        }
      },
    })

    const picture = wrapper.find('pokemonpicture-stub')
    const options = wrapper.find('pokemonoptions-stub')

    expect(picture.exists()).toBeTruthy()
    expect(options.exists()).toBeTruthy()

    expect(picture.attributes('pokemonid')).toBe('5')
    expect(options.attributes('pokemons')).toBeTruthy()
  })
})
