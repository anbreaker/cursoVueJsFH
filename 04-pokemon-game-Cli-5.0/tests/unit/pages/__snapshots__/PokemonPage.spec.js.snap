// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Should match the snapshot 1`] = `
<div>
  <div>
    <h1>Loading...</h1>
  </div>
</div>
`;

exports[`PokemonPage Component Should match the snapshot, when pokemons is loaded 1`] = `
<div>
  <div>
    <h1>Who is this Pokemon?</h1>
    <pokemonpicture-stub pokemonid="1"></pokemonpicture-stub>
    <pokemonoptions-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemonoptions-stub>
    <!---->
  </div>
</div>
`;
